### ISSUE TITLE
Player#setAbsorptionAmount no longer valid

### ISSUE THREAD

**Issue by KioProject123** [PaperSpigot/Paper#9920]
### Expected behavior

Same as 1.20.1
![4f95389dcdb858609406af169a415759](https://github.com/PaperMC/Paper/assets/61594596/c46e8529-cbb8-4625-a376-85ad32865706)

### Observed/Actual behavior

Now at 1.20.2
![d1cbf3d2fc7183c6728c709e1ad412c7](https://github.com/PaperMC/Paper/assets/61594596/5366d1eb-7cea-453e-94b7-11b17eef5a51)

### Steps/models to reproduce

```
    @EventHandler
    public void onPlayerInteract(final @NotNull PlayerInteractEvent e) {
        e.getPlayer().setAbsorptionAmount(10.0);
    }
```

### Plugin and Datapack List

blank

### Paper version

This server is running Paper version git-Paper-280 (MC: 1.20.2) (Implementing API version 1.20.2-R0.1-SNAPSHOT) (Git: 4675152)
You are running the latest version

**Comment by Malfrador** [comment]
Absorption is now an attribute, so it needs to be changed using [getAttribute](https://jd.papermc.io/paper/1.20/org/bukkit/attribute/Attributable.html#getAttribute(org.bukkit.attribute.Attribute)).

The setAbsorptionAmount API methods should likely be deprecated and/or redirected to the attribute, similarly to how it happened with setMaxHealth when an Attribute for that was added years ago.

**Comment by Lulu13022002** [comment]
This is wrong the method works as intended. However the upper bound is now limited by the max_absorption attribute. So you have first to increase this limit (set to 0 by default) to behave like before. But you still need to update the absorption hearts like you do after that. I guess for compatibility this method could either set the attribute before or ignore that limit but this is a feature.

**Comment by Machine-Maker** [comment]
I agree with @Lulu13022002 here. The method absolutely is still useful, but something is needed to address the max absorption attribute. The best thing to do probably depends on what happens if you override the max and just call that `internalSetAbsorptionAmount` method on both entities and Players. If there is some use to it, then perhaps another method with a boolean override. But I don't think the default behavior should be to override the attribute.

**Comment by notTamion** [comment]
imo jdocs should just be changed similar to setHealth: "Sets the entity's health from 0 to getMaxHealth()"
basically just change the amount parameters jdocs to be something like this: "new absorption amount from 0 to its GENERIC_MAX_ABSORPTION Attribute"

**Comment by joestr** [comment]
Funny that after two years this is still not fixed?
